source/extensions/quic_listeners/quiche/envoy_quic_client_stream.cc

```diff
@@ -151,15 +151,21 @@ void EnvoyQuicClientStream::OnInitialHeadersComplete(bool fin, size_t frame_len,
     return;
   }
   quic::QuicSpdyStream::OnInitialHeadersComplete(fin, frame_len, header_list);
-  ASSERT(headers_decompressed() && !header_list.empty());
```

Comment:
Does this really need to be moved down?

Reply:
Yep, the new 11111111...111 test fails that assertion.

Reply:
Does it fail because of header_list is empty? If so, there is nothing to validate at all. I think an early reset makes sense. TBH I don't know where could it fail in QUICHE stack, maybe Qpack decoding? Ideally, QUICHE should reset the stream in that case. 

getResponseStatusNoThrow() uses absl::SimpleAtoi() which doesn't seem to check the status code to be a 3 digit code though. Did I miss something?

Reply:
yeah this is testing for overflow.  Envoy allows greater than 3 digit but not overflowing is important.